### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by tvolkert**
* Allow m4v files in FileType.media filter
* throw if caller tries to use allowedExtensiosn with any file type other than custom

### CODE REVIEW

File: lib/src/file_picker_macos.dart [hunk 1]
```diff
@@ -107,6 +107,10 @@ class FilePickerMacOS extends FilePicker {
   }
 
   String fileTypeToFileFilter(FileType type, List<String>? allowedExtensions) {
+    if (type != FileType.custom && (allowedExtensions?.isNotEmpty ?? false)) {
+      throw Exception(
```
Comment (navaronbracke) [inline lib/src/file_picker_macos.dart:111]: Could you turn this into an `ArgumentError` ? That would make the intent a bit clearer IMO.

And we can tweak the message, to remove the "please" grammar.

```
throw ArgumentError.value(
   allowedExtensions,
   'allowedExtensions',
   'Custom extension filters are only allowed with FileType.custom. '
   'Remove the extension filter or change the FileType to FileType.custom.'
);
```

**Review by navaronbracke — CHANGES_REQUESTED** [review]

File: lib/src/file_picker_macos.dart (hunk 1 above)
Comment (tvolkert) [reply lib/src/file_picker_macos.dart]: Done - thanks for the review!

**Comment by navaronbracke** [comment]
We have one test that is failing, because the expectation needs to be updated.

```
flutter_file_picker/flutter_file_picker/test/file_picker_macos_test.dart: fileTypeToFileFilter() should return the file filter (failed)
  Expected: '"avi", "flv", "mkv", "mov", "mp4", "mpeg", "webm", "wmv", "bmp", "gif", "jpeg", "jpg", "png"'
    Actual: '"avi", "flv", "m4v", "mkv", "mov", "mp4", "mpeg", "webm", "wmv", "bmp", "gif", "jpeg", "jpg", "png"'
     Which: is different.
            Expected: ...  "flv", "mkv", "mov" ...
              Actual: ...  "flv", "m4v", "mkv" ...
                                    ^
             Differ at offset [16](https://github.com/miguelpruivo/flutter_file_picker/actions/runs/7456317251/job/20290515807?pr=1430#step:8:17)
```

**Review by miguelpruivo — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**